File: Dihya/scripts/test_onboarding.js
// Test unitaire Node.js – Onboarding
const axios = require('axios');
(async () => {
  const res = await axios.post('http://localhost:8002/api/onboarding/', { email: 'test@dihya.com', lang: 'fr' });
  if (!res.data.message.includes('Bienvenue')) throw new Error('Pas de bienvenue');
  try {
    await axios.post('http://localhost:8002/api/onboarding/', { lang: 'fr' });
    throw new Error('Erreur RGPD non détectée');
  } catch(e) {
    if (!e.response.data.error.includes('Email')) throw e;
  }
  console.log('Tests onboarding Node.js OK');
})();

File: Dihya/scripts/test_onboarding.py
# Test unitaire Python – Onboarding
import requests

def test_onboarding_ok():
    r = requests.post('http://localhost:8001/api/onboarding/', json={'email': 'test@dihya.com', 'lang': 'fr'})
    assert r.status_code == 200
    assert 'Bienvenue' in r.json().get('message','')

def test_onboarding_rgpd():
    r = requests.post('http://localhost:8001/api/onboarding/', json={'lang': 'fr'})
    assert r.status_code == 400
    assert 'Email' in r.json().get('error','')
